### PR TITLE
Fix the 'Reply' button behavior, align it with Gmail

### LINKED ISSUE #4077: The "reply" button should only reply to the sender, it currently works the same way as "reply all" button
Request by enterprise customer

image

Currently we have reply icon with reply-all functionality which is confusing.

---

Upd: The secure reply icon button triggers the right button, it's the "Reply" button that is broken and works like "Reply all" button.

### PR DESCRIPTION
**Pull request by limonte**
This PR aligns the behavior of the 'Reply' button with Gmail:

- if `from !== acctEmail` => reply to the `from` email address only
- if `from === acctEmail` then reply to all `to` email addresses (excluding `cc` and `bcc`)

Also, it fixes the reply icon button which is injected in Gmail UI, so it now acts the same way as the Secure Reply button.

close #4077

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: test/source/tests/compose.ts [hunk 1]
```diff
@@ -1342,6 +1342,34 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
...
+      const composePage = await ComposePageRecipe.openStandalone(t, browser, 'compatibility', { appendUrl, hasReplyPrompt: true });
+      await composePage.waitAndClick('@action-accept-reply-all-prompt', { delay: 1 });
+      await expectRecipientElements(composePage, { to: ['vladimir@flowcrypt.com'], cc: ['limon.monte@gmail.com'], bcc: ['sweetalert2@gmail.com'] });
+    }));
+
+    ava.default('compose - reply all - from !== acctEmail', testWithBrowser('compatibility', async (t, browser) => {
+      const appendUrl = 'threadId=17d02268f01c7e40&skipClickPrompt=___cu_false___&ignoreDraft=___cu_false___&replyMsgId=17d02268f01c7e40';
+      const composePage = await ComposePageRecipe.openStandalone(t, browser, 'compatibility', { appendUrl, hasReplyPrompt: true });
+      await composePage.waitAndClick('@action-accept-reply-all-prompt', { delay: 1 });
+      await expectRecipientElements(composePage, { to: ['limon.monte@gmail.com', 'vladimir@flowcrypt.com'], cc: ['limon.monte@gmail.com'], bcc: [] });
```
Comment (rrrooommmaaa) [inline test/source/tests/compose.ts:1370]: do we really need `limon.monte@gmail.com` twice?
However, if this mimicks gmail behaviour, then it's ok.

File: test/source/tests/compose.ts [hunk 2]
```diff
@@ -1342,6 +1342,34 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
       await expectRecipientElements(composePage, { to: [], cc: [], bcc: [] });
     }));
 
+    ava.default('compose - reply - from === acctEmail', testWithBrowser('compatibility', async (t, browser) => {
+      const appendUrl = 'threadId=17d02296bccd4c5c&skipClickPrompt=___cu_false___&ignoreDraft=___cu_false___&replyMsgId=17d02296bccd4c5c';
+      const composePage = await ComposePageRecipe.openStandalone(t, browser, 'compatibility', { appendUrl, hasReplyPrompt: true });
+      await composePage.waitAndClick('@encrypted-reply', { delay: 1 });
+      await expectRecipientElements(composePage, { to: ['flowcrypt.compatibility@gmail.com', 'vladimir@flowcrypt.com'], cc: [], bcc: [] });
```
Comment (rrrooommmaaa) [inline test/source/tests/compose.ts:1349]: Strange that `flowcrypt.compatibility@gmail.com` is included in `to` with "reply", whereas it's not included with "reply all". Is there a purpose of such behaviour?

**Review by rrrooommmaaa — CHANGES_REQUESTED** [review]
There are some strangeness in behaviour, I would like to hear your comments.
Also, it would be more straightforwar to name properties `myEmail` and `from` instead of `from` and `orifinalFrom`. Please see the suggetsed update #4111

File: test/source/tests/compose.ts (hunk 2 above)
Comment (limonte) [reply test/source/tests/compose.ts:1349]: My goal here was to align the behavior with Gmail so there will be no surprises for our users.

In this particular case `flowcrypt.compatibility@gmail.com` sends the message 

- `to` 
  - `flowcrypt.compatibility@gmail.com` (itself, `from === acctEmail`)
  - `vladimir@flowcrypt.com`
- `cc`
  - `limon.monte@gmail.com`
- `bcc`
  - `sweetalert2@gmail.com`

Here's the Gmail's behavior which is aligned with the test's expectations:

CleanShot 2021-11-11 at 10 01 55@2x

You can test it by yourself by logging into `flowcrypt.compatibility@gmail.com` and opening this thread https://mail.google.com/mail/u/4/#inbox/WhctKKXHCWWrrFvSmnhdCjDHBtHBbzrqmFPQWKBVqftJdVQFxGtTTPDWZrvZKPXlJpckmNL

File: test/source/tests/compose.ts [hunk 3]
```diff
@@ -1342,6 +1342,34 @@ export const defineComposeTests = (testVariant: TestVariant, testWithBrowser: Te
...
+      const composePage = await ComposePageRecipe.openStandalone(t, browser, 'compatibility', { appendUrl, hasReplyPrompt: true });
+      await composePage.waitAndClick('@encrypted-reply', { delay: 1 });
+      await expectRecipientElements(composePage, { to: ['limon.monte@gmail.com'], cc: [], bcc: [] });
+    }));
+
+    ava.default('compose - reply all - from === acctEmail', testWithBrowser('compatibility', async (t, browser) => {
+      const appendUrl = 'threadId=17d02296bccd4c5c&skipClickPrompt=___cu_false___&ignoreDraft=___cu_false___&replyMsgId=17d02296bccd4c5c';
+      const composePage = await ComposePageRecipe.openStandalone(t, browser, 'compatibility', { appendUrl, hasReplyPrompt: true });
+      await composePage.waitAndClick('@action-accept-reply-all-prompt', { delay: 1 });
+      await expectRecipientElements(composePage, { to: ['vladimir@flowcrypt.com'], cc: ['limon.monte@gmail.com'], bcc: ['sweetalert2@gmail.com'] });
```
[the remark this answers is not in the document]
Comment (limonte) [reply test/source/tests/compose.ts:1363]: In this PR I was focused on fixing the "Reply" button functionality, but you're correct here. There should be `flowcrypt.compatibility@gmail.com` when the user clicks "reply all":

CleanShot 2021-11-11 at 10 06 22@2x

I will fix this in scope of this PR 👍

File: test/source/tests/compose.ts (hunk 1 above)
Comment (limonte) [reply test/source/tests/compose.ts:1370]: Yup, this is aligned with Gmail:

CleanShot 2021-11-11 at 10 07 24@2x

File: test/source/tests/compose.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (limonte) [reply test/source/tests/compose.ts]: Done 👍

**Review by rrrooommmaaa — APPROVED** [review]

**Pull request merged**